fms/models/__init__.py

```diff
@@ -364,20 +446,44 @@ def model_wrap(model):
 
     if len(lazy_sd):
         serialization.load_state_dict_into_model(
-            fms_model,
-            lazy_sd,
-            architecture,
-            source if source is not None else "fms",
-            distributed_strategy,
-            checkpoint_sharding,
-            initial_device,
+            model=fms_model,
+            state_dict=lazy_sd,
+            architecture=architecture,
+            source=source if source is not None else "fms",
+            dtype=data_type_parsed,
+            distributed_strategy=distributed_strategy,
+            checkpoint_sharding=checkpoint_sharding,
+            initial_device=initial_device,
+            rank=rank,
         )
-    elif hasattr(fms_model, "reset_parameters"):
-        fms_model.reset_parameters()
+    else:
+        # move from meta device to real device
+        if initial_device != torch.device("meta"):
+            fms_model.to_empty(device=initial_device)
+        # randomly initialize the model (non-gptq models only)
+        if hasattr(fms_model, "reset_parameters") and not is_gptq:
```

Comment:
are we ever hitting this on GPTQ models?

Reply:
only if a checkpoint is not provided, in which case we should get a gptq model with dummy quantization parameters. Not a use case now but I suppose we want to keep the option open. `reset_parameters` fails on gptq models as it looks for `weight`, so I bypassed it